### PR TITLE
PR editor

### PR DESCRIPTION
**Pull request by thieman**
Alright, this should be fun. :smile: Attempted implementation of #42. I'm junior enough in the Emacs ecosystem that I'm probably committing a handful of unidiomatic sins here, comments and suggestions much appreciated.

This was a bit complex because we're optionally substituting a synchronous flow (`read-input` from minibuffer) with an asynchronous flow (opening up the editor, waiting for user to do their things and jump back into the submitting PR flow). To handle this, I refactored that code to work off of a callback and had the `magit-gh-pulls-pull-editor-finish` function fire off that callback. All state is stored in a buffer local map, which should be sufficient to have multiple editors going at the same time without things getting distorted (though I have not tested this yet).

Still to do:

- [x] Create a minor mode for the editor with proper highlighting and bindings to the `finish` and `quit` commands
- [x] Figure out some better design on the editor to help users figure out what to do
- [x] Refactor various janky elisp-isms if possible, for example the repeated map commands I'm doing right now
- [x] Docstrings for the unfortunate souls to follow

### CODE REVIEW

**Comment by thieman** [comment]
@alexander-yakushev This should be ready for review now whenever you're free, no rush. :+1:

File: magit-gh-pulls.el [hunk 1]
```diff
@@ -80,6 +81,23 @@
 (defvar magit-gh-pulls-collapse-commits t
   "Collapse commits in pull requests listing.")
 
+(defvar-local magit-gh-pulls-previous-winconf nil)
+
+(defvar magit-gh-pulls-editor-mode-map
+  (let ((map (make-keymap)))
+    (define-key map (kbd "C-c C-c") 'magit-gh-pulls-pull-editor-finish)
+    (define-key map (kbd "C-c C-q") 'magit-gh-pulls-pull-editor-quit)
```
Comment (thieman) [inline magit-gh-pulls.el]: My thought here was to use similar bindings from Magit's commit mode since the buffers look fairly similar (I even stole the font-lock formatting from that mode so the summary will be highlighted the same way). The hope is that users will naturally try `C-c C-c` the first time they see this and it'll magically work. The message and easy menu are also there to let them know what the bindings are.

**Comment by alexander-yakushev** [comment]
@thieman I've just tried it and it works! The only possible improvement I see is to replace `C-c C-q` with `C-c C-k` to be consistent with Magit's own bindings. After you do that, I'll be happy to merge this!

**Comment by thieman** [comment]
@alexander-yakushev Shortcut changed and PR squashed/rebased. I kept one extra commit since it's new functionality since your first review; just grabbing the default body for use in full editor. Thanks!

**Comment by thieman** [comment]
Actually I think I misunderstood that a bit, let me try again and get my local running my fork again so I can test it...

**Comment by thieman** [comment]
I seem to have angered GitHub somehow and it's permanently closed the PR, I'll try to open a new one
